### DISCUSSION TITLE
Syntax error with PHP 7.4

### DISCUSSION THREAD

**Discussion by bennoislost** [deployphp/deployer#3974] [Bugs] [answered]
### Deployer Version

v7.5.8

### Target OS

macOS 13.6

### Which PHP version are you using?

PHP 7.4

### Steps to reproduce

Deployer has been installed relative to the project.

Set the PHP CLI version to `7.4.XX`, (`7.4.33` has been used.)

```
composer require  deployer/deployer:7.5.8
```

```
vendor/bin/dep deploy stage=STAGE ---branch=main
```

The deployment fails with the below issue.

```
Parse error: syntax error, unexpected ')', expecting variable (T_VARIABLE) in phar:///{PATH-TO-PROJECT}/vendor/deployer/deployer/bin/dep/vendor/ringcentral/psr7/src/functions.php on line 355
```
(A trialling comma in the function arguments, not valid syntax for PHP 7.4)

Am assuming this is because of a composer version bump in `ringcentral/psr7` as `deployer/deployer:7.5.5` works as expected but I'm struggling to understand how `ringcentral/psr7` is required in the Deployer project.

**Comment by bennoislost** [comment, marked as answer]
Closed as https://github.com/deployphp/deployer/discussions/3925

**Comment by Fab1en** [comment]
I have the same issue here, and I was using version 7.0.0. Upgrading to version 7.5.5 or to 7.5.8 does not solve it.
I think the problem is not in deployer code itself, but in `ringcentral/psr7` as @bennoislost suggested.
The crash occurs for me in github runners, so its difficult to see witch version of `ringcentral/psr7` is used.

Could you please open the ticket again and provide insights about how `ringcentral/psr7` is used by deployer ?

**Reply by antonmedv** [reply]
What PHP version?

**Reply by Fab1en** [reply]
PHP 7.4

**Reply by Fab1en** [reply]
I did more investigations and I'm able to reproduce the problem locally on my computer.
I downloaded the deployer.phar directly from https://deployer.org/download : 

```
$ php deployer-7.5.8.phar deploy preprod -vvv
PHP Parse error:  syntax error, unexpected ')', expecting variable (T_VARIABLE) in phar:///home/vagrant/build/deployer.phar/vendor/ringcentral/psr7/src/functions.php on line 355

$ php deployer-7.0.0.phar deploy preprod -vvv
task deploy:info
[preprod] /usr/bin/php7.4 /home/vagrant/build/deployer-7.0.0.phar worker --port 45243 --task deploy:info --host preprod --decorated -vvv
[preprod] info deploying HEAD
done on preprod
```

So deployer 7.0.0 was working well. The issue comes from a subsequent version.

**Comment by Fab1en** [comment]
I found a workaround: to force an old deployer version in the github action setup:
Add `deployer-version: 7.0.0` to the deployer-php-action configuration.
```
      - name: Setup Deployer
        uses: atymic/deployer-php-action@master
        with:
          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
          ssh-known-hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
          deployer-version: 7.0.0
```

**Comment by Mabahe** [comment]
This is not fixed with [#392](https://github.com/deployphp/deployer/discussions/3925) since it was introduced in 7.5.6 with commit 
https://github.com/deployphp/deployer/commit/44b3a55bd6599985e579e359c759a90a48771f2b

7.5.5 is okay:
https://github.com/deployphp/deployer/blob/v7.5.5/vendor/ringcentral/psr7/src/functions.php#L353

7.5.6 not okay:
https://github.com/deployphp/deployer/blob/v7.5.6/vendor/ringcentral/psr7/src/functions.php#L354

**Comment by Kilbourne** [comment]
Added #4001 to fix

**Comment by mikestreety** [comment]
This is now resolved as of [7.5.11](https://github.com/deployphp/deployer/releases/tag/v7.5.11).

[Note on 3976](https://github.com/deployphp/deployer/discussions/3976#discussioncomment-12121380)
